deob-bytecode/src/main/kotlin/org/openrs2/deob/bytecode/transform/StringDecryptionTransformer.kt

```diff
@@ -0,0 +1,183 @@
+package org.openrs2.deob.bytecode.transform
+
+import com.github.michaelbull.logging.InlineLogger
+import jakarta.inject.Singleton
+import org.objectweb.asm.Opcodes
+import org.objectweb.asm.tree.AbstractInsnNode
+import org.objectweb.asm.tree.FieldInsnNode
+import org.objectweb.asm.tree.FieldNode
+import org.objectweb.asm.tree.LdcInsnNode
+import org.objectweb.asm.tree.MethodInsnNode
+import org.objectweb.asm.tree.TableSwitchInsnNode
+import org.openrs2.asm.classpath.ClassPath
+import org.openrs2.asm.deleteExpression
+import org.openrs2.asm.intConstant
+import org.openrs2.asm.transform.Transformer
+
+@Singleton
+public class StringDecryptionTransformer : Transformer() {
+    private var stringsDecrypted = 0
+    private var classesDecrypted = 0
+    private var stringsInlined = 0
+
+    override fun preTransform(classPath: ClassPath) {
+        for (library in classPath.libraries) {
+            for (clazz in library) {
+                val clinit = clazz.methods.find { it.name == "<clinit>" } ?: continue
+
+                // step 1: identify the decryption methods
+                val inner = clinit.instructions.iterator().asSequence().filterIsInstance<MethodInsnNode>()
```

Comment:
It might be cleaner to use `InsnMatcher.compile("INVOKESTATIC INVOKESTATIC")` in this step.